pkg/nspawntool/run.go

```diff
@@ -199,116 +208,127 @@ func getK8sBindOpts(k8srelease, goPath string) ([]bindOption, error) {
+	Runtime    string
+	bindOpts   []bindOption
+}
 
-	// defaultBindOpts has to be determined after evaluation of cniPath
-	defaultBindOpts := getDefaultBindOpts(cniPath)
-	for _, bo := range defaultBindOpts {
-		outOpt, err := bo.composeBindOption()
-		if err != nil {
-			return nil, err
-		}
-		listBinds = append(listBinds, outOpt)
+func (n *Node) buildBindsListKubernetes(kubeSpawnDir, goPath string) error {
+	k8sBindOpts, err := getK8sBindOpts(n.K8sVersion, kubeSpawnDir, goPath)
+	if err != nil {
+		return err
 	}
+	n.bindOpts = append(n.bindOpts, k8sBindOpts...)
 
-	if kubeSpawnDirParent == "" {
-		kubeSpawnDirParent = parseBind("$PWD")
-	} else if err := utils.CheckValidDir(kubeSpawnDirParent); err != nil {
-		kubeSpawnDirParent = parseBind("$PWD")
+	// NOTE: workaround for making kubelet work with port-forward
+	bo := bindOption{
+		bindPrefix: bindro,
+		srcMount:   path.Join(kubeSpawnDir, "/extras/socat"),
+		dstMount:   "/usr/bin/socat"}
+	n.bindOpts = append(n.bindOpts, bo)
+	return nil
+}
+
+func (n *Node) buildBindsList(kubeSpawnDir string) ([]string, error) {
+	var err error
+
+	if goPath, err = utils.GetValidGoPath(); err != nil {
+		return nil, fmt.Errorf("invalid GOPATH %q", goPath)
+	}
+	if cniPath, err = utils.GetValidCniPath(goPath); err != nil {
+		return nil, fmt.Errorf("invalid CNI_PATH %q", cniPath)
 	}
-	kubeSpawnDir := path.Join(kubeSpawnDirParent, ".kube-spawn")
 
-	if err := os.MkdirAll(kubeSpawnDir, os.FileMode(0755)); err != nil {
-		return nil, fmt.Errorf("unable to create directory %q: %v.", kubeSpawnDir, err)
+	n.bindOpts = append(n.bindOpts, getDefaultBindOpts(kubeSpawnDir, cniPath)...)
+
+	if err := n.buildBindsListKubernetes(kubeSpawnDir, goPath); err != nil {
+		return nil, fmt.Errorf("error getting kubernetes bind mounts list: %v", err)
 	}
 
-	if err := bootstrap.PathSupportsOverlay(kubeSpawnDir); err != nil {
-		return nil, fmt.Errorf("unable to create overlayfs on %q: %v. Try to pass a directory with a different filesystem (like ext4 or XFS) to --kube-spawn-dir.", kubeSpawnDir, err)
+	var runtimeDir string
+	switch n.Runtime {
+	case "docker":
+		runtimeDir = "/var/lib/docker"
+	case "rkt":
+		runtimeDir = "/var/lib/rkt"
+		n.bindOpts = append(n.bindOpts, getRktBindOpts()...)
+	case "crio":
+		return nil, fmt.Errorf("%v runtime not supported", n.Runtime)
 	}
 
 	// mount directory ./.kube-spawn/default/MACHINE_NAME/mount into
-	// /var/lib/docker inside the node
-	mountDir := path.Join(kubeSpawnDir, "default", name, "mount")
+	// /var/lib/{docker, rktlet} inside the node
+	mountDir := path.Join(kubeSpawnDir, "default", n.Name, "mount")
 	if err := os.MkdirAll(mountDir, os.FileMode(0755)); err == nil {
-		bo := bindOption{
-			bindPrefix: bindrw,
-			srcMount:   mountDir,
-			dstMount:   "/var/lib/docker",
-		}
-		outOpt, err := bo.composeBindOption()
-		if err != nil {
-			return nil, err
-		}
-		listBinds = append(listBinds, outOpt)
+		return nil, fmt.Errorf("unable to create dir %q: %v", mountDir, err)
```

Comment:
It must be `err != nil`. Otherwise kube-spawn exits here always, and doesn't run at all.